src/concurrency/transaction.cpp

```diff
@@ -122,16 +116,17 @@ void Transaction::RecordUpdate(const ItemPointer &location) {
       return;
     }
     PL_ASSERT(false);
+  } else {
```

Comment:
It's possible. Just consider the `select_for_update` case. The DBMS locks the version without recording it into read/write set.

Comments added.